### ISSUE TITLE
Return code :241 issue

### ISSUE THREAD

**Issue by toromano** [MatthewPierson/Vieux#13]
hi sir. my mini 2 entered dfu mode finally but new issue:
Device is already in pwned DFU Mode. Not executing exploit.
Exploit worked!
*** SecureROM Signature check remover by Linus Henze ***
Applying patches...
Patches have already been applied. Exiting.

Starting iBSS/iBEC patching
Looks like you are downgrading an iPad Mini 2 to 10.3.3!
Patched iBSS/iBEC
About to re-build IPSW
Entering PWNREC mode...
Getting SHSH...
Restoring...
Note that errors about 'BbSkeyId', 'FDR Client' and 'BasebandFirmware Node' are not important, just ignore them and only report errors that actually stop the restore.
WARNING: Unable to find BbChipID node
ERROR..
Return code: 241
Restore Failed.
Please try again and report the error + full logs if it persists.
Exiting...

pls help me

**Comment by MatthewPierson** [comment]
Please update to the latest version, run it again, and if it fails send me the full log which will now include the full error log

**Comment by toromano** [comment]
"Please update to the latest version"what should I update? and my mini 2 not enter dfu mode again........

**Comment by MatthewPierson** [comment]
Vieux is what you need to update

**Comment by toromano** [comment]
realy sorry for my questions... im a noob .... :( how can update Vieux? reinstall? or on terminal? 
and said again mini 2 not entered dfu mode... why this is soo hard???

**Comment by MatthewPierson** [comment]
Just download the files from this github repo again. Just google the button combination for your iPad to enter dfu mode

**Comment by toromano** [comment]
but my mini2 already entered dfu mode. exploit can't find. 3 hours before started downgrade and give code: 241 issue but now says not entered dfu mode.... why don't enter dfu mode again on terminal?

**Comment by toromano** [comment]
now start downgrade but give same error code 241. I updated Vieux

**Comment by toromano** [comment]
Device is now in pwned DFU Mode.
(12.28 seconds)
Exploit worked!
*** SecureROM Signature check remover by Linus Henze ***
Applying patches...
Successfully applied patches

Starting iBSS/iBEC patching
Looks like you are downgrading an iPad Mini 2 to 10.3.3!
Patched iBSS/iBEC
About to re-build IPSW
Entering PWNREC mode...
Getting SHSH...
Restoring...
Note that errors about 'BbSkeyId', 'FDR Client' and 'BasebandFirmware Node' are not important, just ignore them and only report errors that actually stop the restore.
WARNING: Unable to find BbChipID node
Version: 81b98e0425e17250cc83d5badaf9a8cc6399f481 - 245
Libipatcher version: 3159a387584e352f690cca859e013c3a4683f3e8 - 69
Odysseus support: yes
INFO: device serial number is F9FSCU7KFCM9
[INFO] 64-bit device detected
futurerestore init done
reading signing ticket resources/restoreFiles/apnonce.shsh is done
Found device iPad4,4 j85ap
[TSSC] opening resources/restoreFiles/BuildManifest_iPad4,4.plist
[TSSR] User specified not to request a baseband ticket.
Request URL set to https://gs.apple.com/TSS/controller?action=2
Sending TSS request attempt 1... response successfully received
[Error] failed to read baseband[Error] Fail code=-15
Failed with errorcode=-15

ERROR..
Return code: 241
Restore Failed.
Please try again and report the error/send me the full logs and the 'errorlogrestore.txt' file if it persists
Exiting...

**Comment by Godlike1125** [comment]
I am having the same problem as well.

**Comment by MatthewPierson** [comment]
Please try with the latest commit, it has a potential fix for this issue but I can't test myself as I don't have any A7 iPads (This is the same message I'm replying to all relevant issues :) )

**Comment by MatthewPierson** [comment]
@Godlike1125 @toromano Latest commit *actually* fully fixed all iPad Air and iPad mini 2 issues, please give it a try!

**Comment by kero990** [comment]
Yes, I succeeded, but succeeded in a bunch of errors, which made me frightened. At present, the success rate is not particularly high, and itunes often jumps out and restarts the device, especially when you close itunes. So, it is interesting that in order to avoid iTunes disturbing you, you have to open iTunes instead. Of course, it may not be the problem of itunes, after all, the situation is different every time.

In addition, according to my tests these days, the macos virtual machine under windows can often get the same error results as macbook. Maybe this means that using virtual macos can also succeed, although I ended up with a virtual machine failure and macbook succeeded.

At first I thought that this program was very picky of the data cable. The original data cable would be much better, but in the end the original cable also failed many times, so it felt like a mystery. . .

In short, thank you very much for your efforts, it will allow me to use the iPad mini 2 for another 5 years :)

**Comment by toromano** [comment]
> Yes, I succeeded, but succeeded in a bunch of errors, which made me frightened. At present, the success rate is not particularly high, and itunes often jumps out and restarts the device, especially when you close itunes. So, it is interesting that in order to avoid iTunes disturbing you, you have to open iTunes instead. Of course, it may not be the problem of itunes, after all, the situation is different every time.
> 
> In addition, according to my tests these days, the macos virtual machine under windows can often get the same error results as macbook. Maybe this means that using virtual macos can also succeed, although I ended up with a virtual machine failure and macbook succeeded.
> 
> At first I thought that this program was very picky of the data cable. The original data cable would be much better, but in the end the original cable also failed many times, so it felt like a mystery. . .
> 
> In short, thank you very much for your efforts, it will allow me to use the iPad mini 2 for another 5 years :)

hi. my mini 2 nt entereddfu mode. after restart second try. but entered with before updated... how can you did it?

**Comment by MatthewPierson** [comment]
@toromano I will be closing this issue in 24 hours if you no longer need help/the issue has been resolved. If you still need help then comment with what you need and I will not close it.

**Comment by Godlike1125** [comment]
The commit from 3 days ago totally worked on my iPad. I really appreciate
your effort and time for this community.

**Comment by MatthewPierson** [comment]
Good to hear! Thank you, that means a lot.
